kornia/contrib/diamond_square.py

```diff
@@ -0,0 +1,208 @@
+# based on: https://github.com/anguelos/tormentor/blob/e8050ac235b0c7ad3c7d931cfa47c308a305c486/diamond_square/diamond_square.py  # noqa: E501
+import math
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from kornia.enhance import normalize_min_max
+from kornia.filters import filter2D
+from kornia.testing import KORNIA_CHECK, KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+
+# the default kernels for the diamond square
+default_diamond_kernel: List[List[float]] = [[0.25, 0.0, 0.25], [0.0, 0.0, 0.0], [0.25, 0.0, 0.25]]
+default_square_kernel: List[List[float]] = [[0.0, 0.25, 0.0], [0.25, 0.0, 0.25], [0.0, 0.25, 0.0]]
+
+
+def _diamond_square_seed(
+    replicates: int,
+    width: int,
+    height: int,
+    random_fn: Callable,
+    device: Optional[torch.device] = None,
+    dtype: Optional[torch.dtype] = None,
+) -> Tensor:
+    """Helper function to generate the diamond square image seee.
+
+    Args:
+        replicates: the num of batched replicas for the image.
+        width: the expected image width.
+        height: the expected image height.
+        random_fn: the random function to generate the image seed.
+        device: the torch device where to create the image seed.
+        dtype: the torch dtype where to create the image seed.
+
+    Return:
+        the generated image seed of size Bx1xHxW.
+    """
+    KORNIA_CHECK(width == 3 or height == 3, "Height or Width must be equal to 3.")
+    # TODO(anguelos): can we avoid transposing and passing always fixed size. This will cause issues with onnx/jit
+    transpose: bool = False
+    if height == 3:
+        transpose = True
+        width, height = height, width
```

Comment:
I do not understand this part. Why do we need to transpose when height is 3?

Reply:
Yeah, it’s a bit confusing this part. Didn’t understand at all @anguelos ?